src/main/java/seedu/address/model/role/DesiredCharacteristics.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.model.role;
+
+import java.util.Arrays;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the desired characteristics of a property that will interest a particular buyer.
+ * Individual characteristics are separated by semicolons.
+ */
+public class DesiredCharacteristics {
+    public static final String MESSAGE_CONSTRAINTS = "Desired characteristics can take any value, " +
+            "and can also be left blank.";
+
+    /*
+     * The first character of the desired characteristics must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input (when it should be a null wrapped in an Optional).
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
+
+    public static final String CHARACTERISTIC_DELIMITER = ";";
+
+    public final String[] characteristics;
```

Comment:
Is there a reason you chose to represent the whole characteristics array as one obj, instead of each characteristic being an obj? Seems to make more sense to do it like the latter, similar to how the `tags` field and `tag` class are implemented (this way we can validate each characteristics, not the whole array)

Reply:
I discussed with Chrysline and she mentioned that we would want to store characteristics as a list of strings delimited by ";" - can be changed easily though.